### PR TITLE
internal/ethapi: delete needless error check

### PR DESCRIPTION
**Pull request by tomdever**
delete no use params in testfiles.

### CODE REVIEW

File: internal/ethapi/api_test.go [hunk 1]
```diff
@@ -1244,7 +1244,7 @@ func TestFillBlobTransaction(t *testing.T) {
 			if len(tc.err) > 0 {
 				if err == nil {
 					t.Fatalf("missing error. want: %s", tc.err)
-				} else if err != nil && err.Error() != tc.err {
+				} else if err.Error() != tc.err {
```
Comment (holiman) [inline internal/ethapi/api_test.go:1247]: Why did you remove the `nil`-check?

File: trie/stacktrie_test.go [hunk 2]
```diff
@@ -381,7 +381,7 @@ func TestStacktrieNotModifyValues(t *testing.T) {
 	}
 }
 
-func buildPartialTree(entries []*kv, t *testing.T) map[string]common.Hash {
+func buildPartialTree(entries []*kv) map[string]common.Hash {
```
Comment (holiman) [inline trie/stacktrie_test.go:384]: I think it's ok to have this reference stay. Probably, the person who wrote this once upon a time was debugging something, and maybe used `t.Log` to emit info. Test-functions passing around the `t` doesn't hurt, imo

**Review by holiman — COMMENTED** [review]
Thanks, mostly ok IMO

File: trie/stacktrie_test.go (hunk 2 above)
Comment (tomdever) [reply trie/stacktrie_test.go:384]: Yes, it does not hurt, and can be used for debugging in the future.

File: internal/ethapi/api_test.go (hunk 1 above)
Comment (tomdever) [reply internal/ethapi/api_test.go:1247]: Because line 1245 has checked `if err==nil`, so in `else`, do not need to check `err!=nil`

File: core/blockchain_test.go [hunk 3]
```diff
@@ -2345,7 +2345,7 @@ func testInsertKnownChainData(t *testing.T, typ string, scheme string) {
 			}
 		}
 	} else {
-		inserter = func(blocks []*types.Block, receipts []types.Receipts) error {
+		inserter = func(blocks []*types.Block, _ []types.Receipts) error {
```
Comment (rjl493456442) [inline core/blockchain_test.go:2348]: I feel this change is a bit unnecessary. We have a tons of unused parameters, e.g. for implementing some interfaces.

I am worried people will submit endless prs to change these things.

File: core/blockchain_test.go [hunk 4]
```diff
@@ -2519,7 +2519,7 @@ func testInsertKnownChainDataWithMerging(t *testing.T, typ string, mergeHeight i
...
```
Comment (rjl493456442) [inline core/blockchain_test.go:2522]: same here

File: internal/ethapi/api_test.go [hunk 5]
```diff
@@ -361,7 +361,7 @@ func allTransactionTypes(addr common.Address, config *params.ChainConfig) []txDa
 	}
 }
 
-func allBlobTxs(addr common.Address, config *params.ChainConfig) []txData {
+func allBlobTxs(addr common.Address, _ *params.ChainConfig) []txData {
```
Comment (rjl493456442) [inline internal/ethapi/api_test.go:364]: same here

File: core/blockchain_test.go (hunk 3 above)
Comment (tomdever) [reply core/blockchain_test.go]: yes, it is a bit unnecessary. And test ci has already checked this functions. I have restored this functions.

File: core/blockchain_test.go (hunk 4 above)
Comment (tomdever) [reply core/blockchain_test.go]: yes, it is a bit unnecessary. And test ci has already checked this functions. I have restored this functions.

File: internal/ethapi/api_test.go (hunk 5 above)
Comment (tomdever) [reply internal/ethapi/api_test.go]: yes, it is a bit unnecessary. And test ci has already checked this functions. I have restored this functions.

**Review by rjl493456442 — APPROVED** [review]
